java/clients/src/main/java/sleeper/clients/status/update/RenameTable.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.clients.status.update;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sleeper.configuration.properties.instance.InstanceProperties;
+import sleeper.configuration.properties.table.S3TableProperties;
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.configuration.properties.table.TablePropertiesStore;
+import sleeper.configuration.table.index.DynamoDBTableIndex;
+import sleeper.core.table.TableIdentity;
+import sleeper.core.table.TableIndex;
+import sleeper.core.table.TableNotFoundException;
+import sleeper.core.table.TableWithNameAlreadyExistsException;
+
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+import static sleeper.configuration.utils.AwsV1ClientHelper.buildAwsV1Client;
+
+public class RenameTable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RenameTable.class);
+    private final TableIndex tableIndex;
+    private final TablePropertiesStore tablePropertiesStore;
+
+    public RenameTable(AmazonS3 s3Client, AmazonDynamoDB dynamoDB, InstanceProperties instanceProperties) {
+        this(new DynamoDBTableIndex(instanceProperties, dynamoDB),
+                S3TableProperties.getStore(instanceProperties, s3Client, dynamoDB));
+    }
+
+    public RenameTable(TableIndex tableIndex, TablePropertiesStore tablePropertiesStore) {
+        this.tableIndex = tableIndex;
+        this.tablePropertiesStore = tablePropertiesStore;
+    }
```

Comment:
The behaviour in this class is already implemented in TablePropertiesStore, and partly tested in S3TablePropertiesStoreIT.SaveProperties.shouldUpdateTableName.